cpp/examples/tpch/README.md

```diff
@@ -0,0 +1,63 @@
+# TPC-H Inspired Examples
+
+Implementing the TPC-H queries using `libcudf`. 
+
+## Data Generation
+
+We leverage the data generator (wrapper around official TPC-H datagen) from [Apache Datafusion](https://github.com/apache/datafusion) for generating data in the form of Parquet files. 
+
+### Requirements 
+
+- Rust
+
+### Steps
+
+1. Clone the datafusion repository.
+```bash
+git clone git@github.com:apache/datafusion.git
+```
+
+2. Run the data generator. The data will be placed in a `data/` subdirectory.
+```bash
+cd datafusion/benchmarks/
+./bench.sh data tpch
+
+# for scale factor 10,
+./bench.sh data tpch10
+```
+
+## Executing Queries
+
+### Steps
+
```

Comment:
I would recommend putting any query example specific instructions (like the expected input args, the output results and how to interpret them) in the cpp file examples as well as `cout` them when running the example.
Something like [this](https://github.com/rapidsai/cudf/blob/branch-24.08/cpp/examples/parquet_io/parquet_io.cpp#L80) and [this](https://github.com/rapidsai/cudf/blob/branch-24.08/cpp/examples/parquet_io/parquet_io.cpp#L130)